src/parsers/perf/perfparser.cpp

```diff
@@ -1583,23 +1583,35 @@ void PerfParser::startParseFile(const QString& path)
             emit parsingFinished();
         };
 
-        if (path.endsWith(QLatin1String(".perfparser"))) {
-            QFile file(path);
-            if (!file.open(QIODevice::ReadOnly)) {
-                emit parsingFailed(tr("Failed to open file %1: %2").arg(path, file.errorString()));
-                return;
-            }
+        QFile file(path);
+        if (!file.open(QIODevice::ReadOnly)) {
+            emit parsingFailed(tr("Failed to open file %1: %2").arg(path, file.errorString()));
+            return;
+        }
+
+        if (file.peek(11) == "QPERFSTREAM") {
             d.setInput(&file);
             while (!file.atEnd() && !d.stopRequested) {
                 if (!d.tryParse()) {
-                    emit parsingFailed(tr("Failed to parse file"));
+                    // TODO: provide reason
+                    emit parsingFailed(tr("Failed to parse file %1: %2").arg(path, QStringLiteral("unknown reason")));
```

Comment:
inconsistent capitalization - note the `File format unknown` starts upper case here, but this message and the one in the middle uses lower case.